source_code/main_mcu/src/LOGIC/logic_gui.c

```diff
@@ -329,26 +330,28 @@ void logic_gui_display_login_password_TOTP(child_cred_node_t* child_node)
         prev_gen_credential_flag = TRUE;
     }
     
-    /* User approved, decrypt password and TOTPsecret */
-    logic_encryption_ctr_decrypt(child_node->password, child_node->ctr, MEMBER_SIZE(child_cred_node_t, password), prev_gen_credential_flag);
-    
-    /* 0 terminate password */
-    child_node->cust_char_password[MEMBER_ARRAY_SIZE(child_cred_node_t, cust_char_password) - 1] = 0;
-    
-    /* Check for valid password */
-    if (child_node->passwordBlankFlag != FALSE)
+    if (display_totp_only == FALSE)
     {
-        child_node->cust_char_password[0] = 0;
+        /* User approved, decrypt password and TOTPsecret */
+        logic_encryption_ctr_decrypt(child_node->password, child_node->ctr, MEMBER_SIZE(child_cred_node_t, password), prev_gen_credential_flag);
+        
+        /* 0 terminate password */
+        child_node->cust_char_password[MEMBER_ARRAY_SIZE(child_cred_node_t, cust_char_password) - 1] = 0;
```

Comment:
hmmm this is an interesting one.... not sure why I made it that way.
could you change to child_node->pwdTerminatingZero = 0; ?